src/uct/ib/ud/base/ud_ep.c

```diff
@@ -737,14 +769,23 @@ static void uct_ud_ep_rx_creq(uct_ud_iface_t *iface, uct_ud_neth_t *neth)
                        uct_ib_unpack_uint24(ctl->conn_req.ep_addr.ep_id),
                        ep->dest_ep_id);
 
-    /* creq must always have same psn */
-    ucs_assertv_always(ep->rx.ooo_pkts.head_sn == neth->psn,
-                       "iface=%p ep=%p conn_sn=%d ep_id=%d, dest_ep_id=%d rx_psn=%u "
-                       "neth_psn=%u ep_flags=0x%x ctl_ops=0x%x rx_creq_count=%d",
+    /* Discard duplicate CREQ or CREQ after CREP */
+    if (UCT_UD_PSN_COMPARE(neth->psn, <, ep->rx.ooo_pkts.head_sn)) {
+        uct_ud_ep_rx_ctl_drop_packet(ep, neth);
+        return;
+    }
+
+    /* CREQ must have same psn */
+    ucs_assertv_always((ep->rx.ooo_pkts.head_sn == neth->psn) ||
+                       (ep->flags & UCT_UD_EP_FLAG_CREP_RCVD),
```

Comment:
why needed to add UCT_UD_EP_FLAG_CREP_RCVD condition into this assert?

Reply:
it is leftover